src/services/environment-manager.js

```diff
@@ -68,13 +72,41 @@ function EnvironmentManager(config) {
       to: toEnvironmentId,
     };
 
-    return agent
+    const deploymentRequestResponse = await agent
       .post(`${config.serverHost}/api/deployment-requests`)
       .set('Authorization', `Bearer ${authToken}`)
       .set('forest-origin', 'Lumber')
       .set('forest-project-id', config.projectId)
-      .send(DeploymentRequestSerializer.serialize(deploymentRequest))
-      .then(() => true);
+      .send(DeploymentRequestSerializer.serialize(deploymentRequest));
+
+    const jobId = deploymentRequestResponse.body.meta.job_id;
+
+    cli.action.start('Copying layout...');
+
+    const checkState = async function checkState() {
+      const jobResponse = await agent
+        .get(`${config.serverHost}/api/jobs/${jobId}`)
+        .set('Authorization', `Bearer ${authToken}`)
+        .set('forest-origin', 'Lumber')
+        .set('forest-project-id', config.projectId);
+
+      if (jobResponse
+          && jobResponse.body
+          && jobResponse.body.state
+          && jobResponse.body.state !== 'inactive'
+          && jobResponse.body.state !== 'active') {
+        cli.action.stop();
+        if (jobResponse.body.state === 'failed') {
```

Comment:
Just a thinking, we should track that to see immediately the issues. Like this should not happened

Reply:
Do we still need to improve this right now regarding the latest errors handling changes we applied ?